### ISSUE TITLE
Develop Frontend Access for Displaying All Resources via API Endpoint

### ISSUE THREAD

**Issue by ank1traj** [WebSorcery/cucoders#1]
The goal of this task is to enable frontend access to an API endpoint that retrieves and displays all available resources within the application. By implementing this feature on the frontend, users will gain visibility and access to a comprehensive view of available resources through the application interface.

**Comment by singhcoder694** [comment]
@ank1traj I am facing problem in setup for this repo... Can you help me in this
<img width="440" alt="Screenshot 2023-12-11 093227" src="https://github.com/WebSorcery/cucoders/assets/114601400/4ba481ac-99ce-4c5e-9604-ef65e898dff7">
<img width="448" alt="Screenshot 2023-12-11 093237" src="https://github.com/WebSorcery/cucoders/assets/114601400/1e5dde6c-bd3d-4de2-9c5c-cc580b8bc5fd">

**Comment by ank1traj** [comment]
@singhcoder694 What's your node version? Please try with latest node and let me know is it working or not?

**Comment by singhcoder694** [comment]
Nodev v20 @ank1traj

**Comment by ank1traj** [comment]
Are you trying `npm install`? If yes please try with `yarn install `. And let me know the status.

**Comment by singhcoder694** [comment]
No I did yarn install

**Comment by ank1traj** [comment]
This issue is coming from open SSL. Please run the `yarn install` with `--openssl-legacy-provider`

**Comment by singhcoder694** [comment]
Is there any need to use any specific yarn version I am using Yarn v3

**Comment by singhcoder694** [comment]
yarn install --openssl-legacy-provider, showing this error:
Unknown Syntax Error: Unsupported option name ("--openssl-legacy-provider").

**Comment by singhcoder694** [comment]
After cloning project:
I ran yarn install command but this was the first error:

Usage Error: The nearest package directory (C:\Users\91620\cucoders) doesn't seem to be part of the project declared in C:\Users\91620.

- If C:\Users\91620 isn't intended to be a project, remove any yarn.lock and/or package.json file there.
- If C:\Users\91620 is intended to be a project, it might be that you forgot to list cucoders in its workspace configuration.
- Finally, if C:\Users\91620 is fine and you intend cucoders to be treated as a completely separate project (not even a workspace), create an empty yarn.lock file in it.

If you can tell me what to do in this case as I have tried many solutions😔

**Comment by ank1traj** [comment]
I've yarn 1.22.19 and it's working fine. Please try this once. If it will not work then we can do a meet and figure out the issues.

**Comment by singhcoder694** [comment]
Yes now it is done but only changing to this version didn't work, I had to change package.json file also to run the server correctly.

**Comment by singhcoder694** [comment]
Finally it has worked but I had to do bit more changes other than changing version....

I want to work on this issue but unable to understand it properly if you can provide some more reference and explanation..

**Comment by singhcoder694** [comment]
<img width="934" alt="Screenshot 2023-12-13 102843" src="https://github.com/WebSorcery/cucoders/assets/114601400/3adb8f0b-8d52-4452-92b2-59d4eea9f350">
 
On resource page it is showing only this page nothing else, and when trying to login it is showing undefined on local and on official website also it is not working https://main.cuchapter.tech/resources it leads to home page again if we try to login.

**Comment by ank1traj** [comment]
You can turn off the authentication code from the `App.js` file.

**Comment by singhcoder694** [comment]
Hmm I got what to do, we need to use api call to fetch data from cloudinary and drive and using forEach loop we will show data instead of hardcode, But how am I going to get the data which I have to fetch since there is no link to get data from cloudinary or drive

**Comment by ank1traj** [comment]
Provide me with your email ID. Will give you the access of the same

**Comment by singhcoder694** [comment]
vivekkr694@gmail.com

**Comment by ank1traj** [comment]
Added

**Comment by singhcoder694** [comment]
Ok , thankyou

**Comment by singhcoder694** [comment]
curl https://<API_KEY>:<API_SECRET>@api.cloudinary.com/v1_1/<CLOUD_NAME>/resources/image 
Nedd API KEY and API SECRET but unable to get it as it is showing no permission

**Comment by ank1traj** [comment]
You can get it from the Cloudinary dashboard only.

**Comment by singhcoder694** [comment]
In my case it is not in setting tab may be I don't have permission or something else...
@ank1traj 
<img width="960" alt="Screenshot 2023-12-14 143130" src="https://github.com/WebSorcery/cucoders/assets/114601400/8bf596c7-96c2-4f93-b309-a272a6051c38">

**Comment by ank1traj** [comment]
Updated the role. Please try now

**Comment by singhcoder694** [comment]
@ank1traj the data retrieved from API is like this...
{
      asset_id: '43797c36ffa399ffbd807fb520c4582a',
      public_id: 'addResources/Database/1656485202797_mw9pvc',
      format: 'pdf',
      version: 1656485207,
      resource_type: 'image',
      type: 'upload',
      created_at: '2022-06-29T06:46:47Z',
      bytes: 960146,
      width: 595,
      height: 841,
      backup: true,
      folder: 'addResources/Database',
      url: 'http://res.cloudinary.com/cuchapter/image/upload/v1656485207/addResources/Database/1656485202797_mw9pvc.pdf',
      secure_url: 'https://res.cloudinary.com/cuchapter/image/upload/v1656485207/addResources/Database/1656485202797_mw9pvc.pdf'
    }
As you can see it is missing many aspects such as Image URL , Category, Title etc. If you may tell what to specify if these values are not known.

**Comment by ank1traj** [comment]
`https://res.cloudinary.com/cuchapter/image/upload/v1656485437/addResources/Data-Structure-Algorithms/Algorithms_1_x8d2xx.pdf`

This is the public url and some data are static and some are dynamic which is coming from the API itself. You can work in this way.

**Comment by singhcoder694** [comment]
@ank1traj I have solved this issue but unable to check due to blank page error, if you may tell me reason for this....
<img width="959" alt="Screenshot 2023-12-23 133658" src="https://github.com/WebSorcery/cucoders/assets/114601400/c2f2d1d3-8bb4-43b6-9031-edc59e865c18">

**Comment by ank1traj** [comment]
@singhcoder694 What is the issue here? Is there any console error?

**Comment by singhcoder694** [comment]
There are so many errors in console, due to process.env, but I don't think it is reason for blank screen @ank1traj 
<img width="355" alt="Screenshot 2023-12-24 103404" src="https://github.com/WebSorcery/cucoders/assets/114601400/ed8121ff-c27a-4228-92bd-118de483eab3">

**Comment by ank1traj** [comment]
It's error from cloudinary. Please check your code where you're using split function. And, please let me know the result.

**Comment by singhcoder694** [comment]
@ank1traj , sir I just deleted my last clone and installed again but again it is looking same, is this working fine at your side.... Due to this many people wouldn't be able to contribute😣

**Comment by ank1traj** [comment]
It's working fine for me @singhcoder694. Let's connect over meet if it's still not working for you.

**Comment by singhcoder694** [comment]
@ank1traj I found the problem, I think due to installation of cloudinary package-lock.json also initiated in my project, maybe that was overlapping with yarn.lock creating error.

Finally hope I would be able to make PR for this by tommorow 🙂, I have put my alot of time in configuring mistakes and understanding API😣

**Comment by ank1traj** [comment]
Great. Please don't use multiple package installer in same repo. Also, whatever you put all will go to your skill sets and help you in longer run.

**Comment by singhcoder694** [comment]
@ank1traj can you help me in this issue, I have tried everything but unable to complete it😔 if you can provide help in this...

**Comment by ank1traj** [comment]
@singhcoder694 Sure, What help you required?

**Comment by singhcoder694** [comment]
Can we connect on google meet to discuss it tommorow? If you have time?

**Comment by ank1traj** [comment]
Sure

**Comment by singhcoder694** [comment]
So how can I contact you? Any contact information?

**Comment by ank1traj** [comment]
https://calendly.com/honesthacker

**Comment by singhcoder694** [comment]
Done @ank1traj
